libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable import/order */
+import { join } from "path";
+import { ApiGatewayMethod } from "@cdktf/provider-aws/lib/api-gateway-method";
+import { ApiGatewayResource } from "@cdktf/provider-aws/lib/api-gateway-resource";
+import { ApiGatewayRestApi } from "@cdktf/provider-aws/lib/api-gateway-rest-api";
+import { ApiGatewayDeployment } from "@cdktf/provider-aws/lib/api-gateway-deployment";
+import { ApiGatewayStage } from "@cdktf/provider-aws/lib/api-gateway-stage";
+import { ApiGatewayMethodSettings } from "@cdktf/provider-aws/lib/api-gateway-method-settings";
+import { LambdaPermission } from "@cdktf/provider-aws/lib/lambda-permission";
+import { DataAwsRegion } from "@cdktf/provider-aws/lib/data-aws-region";
+import { DataAwsCallerIdentity } from "@cdktf/provider-aws/lib/data-aws-caller-identity";
+
+import { Construct } from "constructs";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { convertBetweenHandlers } from "../convert";
+import { Code } from "../core/inflight";
+import { Function } from "./function";
+import { ApiGatewayIntegration } from "@cdktf/provider-aws/lib/api-gateway-integration";
+import { IResource } from "../core";
+
+/**
+ * AWS Implementation of `cloud.Api`.
+ */
+export class Api extends cloud.ApiBase {
+  private readonly api: ApiGatewayRestApi;
+  constructor(scope: Construct, id: string, props: cloud.ApiProps = {}) {
+    super(scope, id, props);
+    this.api = new ApiGatewayRestApi(this, "api", {
+      name: id,
```

Comment:
This name should be generated using the ResourceNames utility

Reply:
I didn't know that there is helper for that. 
I'll start with a simple pattern, because I could't find any docs about RestApi name limitations in AWS.